lib/ansible/modules/monitoring/zabbix_proxy.py

```diff
@@ -0,0 +1,369 @@
+        required: false
+        default: {}
+'''
+
+EXAMPLES = '''
+- name: Create a new proxy or update an existing proxies info
+  local_action:
+    module: zabbix_proxy
+    server_url: http://monitor.example.com
+    login_user: username
+    login_password: password
+    proxy_name: ExampleProxy
+    description: ExampleProxy
+    status: 5
+    state: present
+    interface:
+        type: 0
+        main: 1
+        useip: 1
+        ip: 10.xx.xx.xx
+        dns: ""
+        port: 10050
+'''
+
+RETURN = ''' # '''
+
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from zabbix_api import ZabbixAPI
+
+    HAS_ZABBIX_API = True
+except ImportError:
+    HAS_ZABBIX_API = False
+
+
+class Proxy(object):
+    def __init__(self, module, zbx):
+        self._module = module
+        self._zapi = zbx
+        self.existing_data = None
+
+    def proxy_exists(self, proxy_name):
+        result = self._zapi.proxy.get({
+            'output': 'extend', 'selectInterface': 'extend',
+            'filter': {'host': proxy_name}})
+
+        if len(result) > 0 and 'proxyid' in result[0]:
+            self.existing_data = result[0]
+            return result[0]['proxyid']
+        else:
+            return result
+
+    def add_proxy(self, data):
+        try:
+            if self._module.check_mode:
+                self._module.exit_json(changed=True)
+
+            parameters = {}
+            for item in data:
+                if data[item]:
+                    parameters[item] = data[item]
+
+            proxy_ids_list = self._zapi.proxy.create(parameters)
+            self._module.exit_json(changed=True,
+                                   result="Successfully added proxy %s (%s)" %
+                                          (data['host'], data['status']))
+            if len(proxy_ids_list) >= 1:
+                return proxy_ids_list['proxyids'][0]
+        except Exception as e:
+            self._module.fail_json(msg="Failed to create proxy %s: %s" %
+                                   (data['host'], e))
+
+    def delete_proxy(self, proxy_id, proxy_name):
+        try:
+            if self._module.check_mode:
+                self._module.exit_json(changed=True)
+            self._zapi.proxy.delete([proxy_id])
+            self._module.exit_json(changed=True,
+                                   result="Successfully deleted" +
+                                          " proxy %s" % proxy_name)
+        except Exception as e:
+            self._module.fail_json(msg="Failed to delete proxy %s: %s" %
+                                       (proxy_name, str(e)))
+
+    def compile_interface_params(self, new_interface):
+        old_interface = {}
+        if 'interface' in self.existing_data and \
+           len(self.existing_data['interface']) > 0:
+                old_interface = self.existing_data['interface']
+
+        final_interface = old_interface.copy()
+        final_interface.update(new_interface)
+
```

Comment:
`final_interface = dict((k,str(v)) for k,v in final_interface.iteritems())`

seems to be required here since the values from new_interface may contain non-strings while the values returned by the API are always strings. With this added line, the module becomes idempotent with passive proxies.